### ISSUE TITLE
[web] TextField inside TableRow without TableCell wrapper does not accept input on web when semantics are enabled

### ISSUE THREAD

**Issue by liubou-s-inno** [flutter/flutter#170575]
#### Description

On Flutter Web, when a `TextField` is placed directly inside a `TableRow` (i.e., not wrapped in a `TableCell`), the text field does not accept any input when accessibility semantics are enabled. The text field appears focusable: the caret is visible, and the field gains focus as expected. However, when the user types, no text appears in the field.

#### Steps to Reproduce

1. Create a `Table` with a `TableRow` containing a `TextField` directly (not wrapped in `TableCell`).
2. Enable semantics with `SemanticsBinding.instance.ensureSemantics()`.
3. Attempt to focus and type into the field.

#### Workarounds

- Wrapping the `TextField` in a `TableCell` resolves the issue.

However, the documentation for `TableRow.children` states:

> Children may be wrapped in `TableCell` widgets to provide per-cell configuration to the `Table`, but children are not required to be wrapped in `TableCell` widgets.

This suggests that `TableCell` wrapping should be optional. 

### Expected results

The `TextField` should accept input as normal, regardless of whether it is wrapped in a `TableCell`.

### Actual results

On web, the `TextField` gains focus but does not respond to keyboard input when not wrapped in a `TableCell`.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/semantics.dart';

void main() {
  WidgetsFlutterBinding.ensureInitialized();
  SemanticsBinding.instance.ensureSemantics();

  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Table(
          children: [
            TableRow(
              children: [
                TableCell(
                  child: TextField(),
                ),
              ],
            ),
          ],
        ),
      ),
    );
  }
}
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.32.3, on macOS 14.6.1 23G93 darwin-arm64, locale en-US) [519ms]
    • Flutter version 3.32.3 on channel stable at /Users/user/Developer/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5c1433509f (2 days ago), 2025-06-11 13:12:24 -0700
    • Engine revision 31c4875c7a
    • Dart version 3.8.1
    • DevTools version 2.45.1

[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0) [1,837ms]
    • Android SDK at /Users/user/Library/Android/sdk
    • Platform android-35, build-tools 35.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
      This is the JDK bundled with the latest Android Studio installation on this machine.
      To manually set the JDK path, use: `flutter config --jdk-dir="path/to/jdk"`.
    • Java version OpenJDK Runtime Environment (build 21.0.4+-12422083-b607.1)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.4) [1,061ms]
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15F31d
    • CocoaPods version 1.16.2

[✓] Chrome - develop for the web [30ms]
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.2) [29ms]
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.4+-12422083-b607.1)

[✓] IntelliJ IDEA Community Edition (version 2024.2.3) [28ms]
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] Connected device (3 available) [11.4s]
    • iPhone 15 Pro Max (mobile) • 8C684370-7222-4C78-A294-767809F4010E • ios            • com.apple.CoreSimulator.SimRuntime.iOS-17-5 (simulator)
    • macOS (desktop)            • macos                                • darwin-arm64   • macOS 14.6.1 23G93 darwin-arm64
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 137.0.7151.104

[✓] Network resources [663ms]
    • All expected network resources are available.

• No issues found!

```

</details>

**Comment by tirth-patel-nc** [comment]
# Triage Report

I was able to reproduce the issue on Web using the provided code sample and steps (placing a `TextField` inside a `TableRow` without wrapping it in a `TableCell`) on `stable (3.32.4)` and `master (3.33.0-1.0.pre.494)`. However, it works as expected on `stable (3.29.3)`. Hence labelling it as a regression.

This issue does not occur on Android, iOS, or macOS.

Please check the video below demonstrating the issue. I tested it on Chrome, Safari, and Firefox. On Firefox and Safari, I was able to type a few characters, while on Chrome, I couldn't type anything. However, the issue is present across all three browsers.

https://github.com/user-attachments/assets/3766c383-ecb3-4b38-bc08-3284ef49006b

**Comment by mdebbar** [comment]
FYI @chunhtai. Any idea what may be happening?

**Comment by chunhtai** [comment]
it is possible that the tableCell's semantics somehow messes the textfield semantics. will need to see the semantics tree dump to be sure

**Comment by chunhtai** [comment]
cc @hannah-hyj who added table semantics

**Comment by pedromassango** [comment]
>it is possible that the tableCell's semantics somehow messes the textfield semantics. will need to see the semantics tree dump to be sure

From my investigation, yes, could be caused by `Semantics`, there are some related issues cause by semantics, you can pretty much reproduce this with just a simple `TextInputField`.

**Comment by hannah-hyj** [comment]
The semantics tree dump looks the same with or without the table cell.

Table overrides assembleSemanticsNode and will insert a table cell semantics node in between even if not wrapped by TableCell. 

I'm not sure why the textfield can get focus but can't get input in this case.


```
SemanticsNode#0
 │ Rect.fromLTRB(0.0, 0.0, 2016.0, 1366.0)
 │
 └─SemanticsNode#1
   │ Rect.fromLTRB(0.0, 0.0, 1008.0, 683.0) scaled by 2.0x
   │ textDirection: ltr
   │
   └─SemanticsNode#2
     │ Rect.fromLTRB(0.0, 0.0, 1008.0, 683.0)
     │ sortKey: OrdinalSortKey#a931b(order: 0.0)
     │
     └─SemanticsNode#3
       │ Rect.fromLTRB(0.0, 0.0, 1008.0, 683.0)
       │ flags: scopesRoute
       │
       └─SemanticsNode#4
         │ Rect.fromLTRB(0.0, 0.0, 1008.0, 48.0)
         │ role: table
         │
         └─SemanticsNode#7
           │ Rect.fromLTRB(0.0, 0.0, 1008.0, 48.0)
           │ role: row
           │ indexInParent: 0
           │
           └─SemanticsNode#5
             │ Rect.fromLTRB(0.0, 0.0, 1008.0, 48.0)
             │ role: cell
             │ indexInParent: 0
             │
             └─SemanticsNode#6
                 Rect.fromLTRB(0.0, 0.0, 1008.0, 48.0)
                 actions: didGainAccessibilityFocus, didLoseAccessibilityFocus,
                   focus, tap
                 flags: isTextField, hasEnabledState, isEnabled, isFocusable
                 textDirection: ltr
                 currentValueLength: 0
                 inputType: text
```

**Comment by hannah-hyj** [comment]
I think this is related to my former table PR, if i set `addCellWrapper` to `false` in table `assembleSemanticsNode` the issue will disappear, I will look into this.
